They mostly come at night, mostly...

jasonmccallister opened the pull request:
Lots of little changes. Mostly views and moving the repo to a library.

Waiting for the decision on Twig and I will move the entire mockup into actual code and start making magic happen"

For your morning reading, the [title reference](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0CCsQtwIwAA&url=http%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DB436avtEXzs&ei=ahpaU4DOEISHyAT5rYHoCQ&usg=AFQjCNEgPZEXhk5O7yWReGLyxQdO_dowjg&sig2=g3uKDkI4hc19aff4DKT2DA&bvm=bv.65397613,d.aWw).

jasonmccallister committed:
route utils

jasonmccallister committed:
Merge branch 'development' of https://github.com/sdphp/grouptopics.org into development

jasonmccallister committed:
Add .idea to gitignore

jasonmccallister committed:
Move Account to Grouptopics\Libraries

jasonmccallister committed:
Add base css and elements for macking the site

jasonmccallister committed:
Add account/settings and profile to routes.php

jasonmccallister committed:
Add basic views - REALLY crappy CSS, only for small mockup, will rewrite soon

jasonmccallister committed:
Update namespace, add docblocks, add accountSettings and showAccountById

jasonmccallister committed:
Add Twig to composer.json and app/config/app.php

jasonmccallister committed:
Here comes the Twig

jasonmccallister committed:
Clean up routes, added names to all routes

jasonmccallister committed:
Add named routes to returns, remove docblocks until finalized

jasonmccallister committed:
Move to blade again

cryptiklemur commented on .gitignore:
```diff
@@ -14,3 +14,4 @@ puphpet
 Vagrantfile
 .vagrant
 tags
+.idea
```
http://stackoverflow.com/questions/7335420/global-git-ignore

Kinda mixed thoughts on this one, but a lot of people i talk to, say IDE specific files (`.idea`) should be in the global gitignore, instead of here

jasonmccallister replied:
I can pull that out. I did wonder about node_modules dir as well. Is that needed?

cryptiklemur replied:
Depends on the project, if the project requires the npm stuff to work, i put it in the project gitignore, if its developer specific, i put it in the global

cryptiklemur commented on app/Grouptopics/Libraries/Account.php:
```diff
@@ -0,0 +1,96 @@
+     * @return mixed
+     */
+    public function getAccountByUserId($id)
+	{
```
Lets switch all these to spaces instead of tabs. Everywhere

jasonmccallister replied:
4 spaces correct?

cryptiklemur replied:
yup

cryptiklemur commented on app/Grouptopics/Libraries/Account.php:
```diff
@@ -0,0 +1,96 @@
+class Account
+{
+    /**
+     * @param $id
```
Not entirely necessary, but its useful to typehint these params. and i think `@return` should be `User`

cryptiklemur commented on app/Grouptopics/Libraries/Account.php:
```diff
@@ -0,0 +1,96 @@
+
+		$user = new User;
+
+		$rules = array(
```
@sdphp/owners What php version are we requiring? This could be switched to `[` and `]`, if we are on 5.4

cryptiklemur commented on app/Grouptopics/Libraries/Account.php:
```diff
@@ -0,0 +1,96 @@
+	{
+		$input = Input::all();
+
+		$attempt = Auth::attempt([
```
if we arent on 5.4, this array syntax will have to change

cryptiklemur commented on app/Grouptopics/Libraries/Account.php:
```diff
@@ -0,0 +1,96 @@
+			'password' 	=> $input['password']
+		]);
+
+		if ($attempt) return Redirect::intended('/account');
```
always expand if statements.

``` php
if ($attempt) {
    return Redirect::intended('/account');
}
```

cryptiklemur commented on app/Grouptopics/Libraries/Account.php:
```diff
@@ -0,0 +1,96 @@
+	{
+		// TODO - make this work!
+	}
+}
```
Make a newline at the end of your files

cryptiklemur commented on app/controllers/AccountController.php:
```diff
@@ -1,48 +1,88 @@
 <?php
```
Namespace?

jasonmccallister replied:
Are you suggesting that this be namespaced?

cryptiklemur replied:
yeah

jasonmccallister replied:
How should I namespace the controller? 

``` php
<?php namespace Grouptopics;
```

cryptiklemur replied:
_probably_ `<?php namespace Grouptopics\Controller;`

depends what the filesystem looks like. It should match

jasonmccallister replied:
The Controller is in the default location for Laravel (app/controllers).. Still requires namespacing?

cryptiklemur replied:
Uh.... @Kindari, input?

ericvanjohnson replied:
I believe @themccallister is correct, this should not require namespacing.

cryptiklemur replied:
Well, the app should probably be namespaced more, i'd actually think it'd be better to move this stuff into the `Grouptopics` directory, and make a dir structure there, that follows PSR-0.

jasonmccallister replied:
I left as is until we hear otherwise. Thanks @aequasi and @shocm.

Also, I hope you like movie references.. I have decided that every pull request's title will be a movie reference from here on out ;)

jasonmccallister replied:
@aequasi I do agree, I wasn't sure if that would apply to controllers. I would like to discuss the dir structure in depth... So once this is merged does someone want to stage the dirs using proper naming standards?

Also, **incredibly** OCD of myself... I wanted to arrange all methods alphabetically.. Thoughts?

cryptiklemur replied:
I've usually seen, and try to follow this myself, methods organized mostly by visibility, then order of usage, public methods should be at the top of the class.

jasonmccallister replied:
Awesome, will keep that in mind. Thanks!

cryptiklemur commented on app/controllers/AccountController.php:
```diff
@@ -1,48 +1,88 @@
 {
-	protected $account;
+    /**
+     * @var Grouptopics\Libraries\Account
```
Should put the whole FQN up in the use statement

cryptiklemur commented on app/controllers/AccountController.php:
```diff
@@ -1,48 +1,88 @@
 
-	public function index()
+    /**
+     * @return \Illuminate\View\View
```
Make a use statement

cryptiklemur commented on app/routes.php:
```diff
@@ -2,23 +2,43 @@
-	dd($users);
 
+// GROUPS requests requiring Auth
+Route::group(array('before' => 'auth'), function()
```
needs to be separated more

``` php
Route::group(
    array('before' => 'auth'),
    function() {
        Route::get('/account', 'AccountController@account');
        Route::get('/account/settings', 'AccountController@accountSettings');
    }
);
```

jasonmccallister replied:
Have that in the local version as well. Will update soon. I went with this:

```
Route::get('/profile/{username}', array(
    'as' => 'profile.view',
    'uses' =>'AccountController@profileView'
));

Route::group(array(
    'before' => 'auth'
), function()
{
    Route::get('/account', array(
        'as' => 'auth.account',
        'uses' => 'AccountController@account'
    ));

    Route::get('/account/settings', array(
        'as' => 'auth.account.settings',
        'uses' => 'AccountController@accountSettings'
    ));
});
```

Thoughts?

cryptiklemur replied:
as soon as you span multiple lines, everything needs to be made multi-line.

``` php
Route::get(
    '/profile/{username}', 
    array(
        'as' => 'profile.view',
        'uses' =>'AccountController@profileView'
    )
);

Route::group(
    array(
        'before' => 'auth'
    ),
    function() {
        Route::get(
            '/account', 
            array(
                'as' => 'auth.account',
                'uses' => 'AccountController@account'
            )
        );
        Route::get(
            '/account/settings',
            array(
                'as' => 'auth.account.settings',
                'uses' => 'AccountController@accountSettings'
            )
        );
    }
);
```

cryptiklemur commented on app/routes.php:
```diff
@@ -2,23 +2,43 @@
+
+
+// GROUPS request that require admin privileges
+Route::group(array('prefix' => 'manage'), function()
```
Same as above

cryptiklemur commented on app/routes.php:
```diff
@@ -2,23 +2,43 @@
+
+
+// JUNK - for testing specific functions as needed
+Route::get('utils', function()
```
Same as above

cryptiklemur commented on app/routes.php:
```diff
@@ -2,23 +2,43 @@
+// GROUPS request that require admin privileges
+Route::group(array('prefix' => 'manage'), function()
+{
+    Route::get('manage', function()
```
Same as above

jasonmccallister committed:
Remove twig templates

jasonmccallister committed:
PSR it up

Convert to spaces as well.

jasonmccallister committed:
PSR it up

Convert to spaces

jasonmccallister committed:
Remove .idea from gitignore

jasonmccallister committed:
Add TODO

Add exceptions for existing emails

jasonmccallister committed:
Back to Blade

jasonmccallister commented:
Updated all but one (namespacing) of the comments.

cryptiklemur commented on app/Grouptopics/Libraries/Account.php:
```diff
@@ -0,0 +1,84 @@
+<?php namespace Grouptopics\Libraries;
```
So, i guess i'll note here. This is a really weird way to do write a namespace, but its laravels standard. Usually, it goes on its own blank line, so the license can go in between `<?php` and the namespace

jasonmccallister replied:
That is in PSR-2, so I think we should correct it.

cryptiklemur commented on app/Grouptopics/Libraries/Account.php:
```diff
@@ -0,0 +1,84 @@
+        $rules = array(
+            'name'   	=> 'required|min:5|max:125',
+            'email'     => 'required|email',
+            'password' 	=> 'required|alpha_num|confirmed|min:8'
```
Align the values

cryptiklemur commented on app/Grouptopics/Libraries/Account.php:
```diff
@@ -0,0 +1,84 @@
+
+        if ($validator->passes()) {
+
+        	$user->name = Input::get('name');
```
spaces are a big weird in this

cryptiklemur commented on app/Grouptopics/Libraries/Account.php:
```diff
@@ -0,0 +1,84 @@
+    {
+        $input = Input::all();
+
+        $attempt = Auth::attempt([
```
multiline this more

cryptiklemur replied:
Did you update this @themccallister?

jasonmccallister replied:
Yes, in 255ded2afa2fe7043a85c2b4554869f712f78f0c I multi-lined the following:

```
return Redirect::route('signup')
    ->withErrors($validator);
```

cryptiklemur replied:
I meant this array in 

``` php
        $attempt = Auth::attempt([
              'email'    => $input['email'],
             'password' => $input['password']
         ]);
```

Should be

``` php
$attempt = Auth::attempt(
    [
        'email'        => $input['email'],
        'password' => $input['password']
    ]
);
```

jasonmccallister replied:
Ahh, hows this? c3a8d36

cryptiklemur replied:
:thumbsup:

jasonmccallister committed:
PSR-2

Align and move namespace to its own line.

jasonmccallister committed:
Multi-line the Redirect

Remove tabs

jasonmccallister committed:
Remove node_modules from gitignore, closes sdphp/grouptopics.org#65

jasonmccallister committed:
Modify Gulp

Modify gulpfile.js to point to new directory “/less”.
Create “/less/grouptopics.less” which imports bootstrap.less.
Update readme to detect language in code block.

jasonmccallister committed:
Remove old CSS files

jasonmccallister committed:
Add permissions, suspended and username to users table

jasonmccallister committed:
Update css

ericvanjohnson commented:
What is the status of this pull request? @aequasi / @themccallister  are we still working on things or is this ready to be merged?

jasonmccallister committed:
Add permissions to $hidden

jasonmccallister committed:
Update CSS and LESS

jasonmccallister commented:
I hit everything he pointed out, I'm going to cover a few more items and then push and update. I will comment again when ready.

jasonmccallister committed:
Update navigation

jasonmccallister committed:
Change to extends account

jasonmccallister committed:
Create account layout

jasonmccallister committed:
Rename user profile method

jasonmccallister committed:
Add getAccountByUsername to account library

jasonmccallister committed:
Remove functionality from account_settings

jasonmccallister commented:
Thats all I have so far, stuck on a few items I'm working out locally.

ericvanjohnson commented:
@themccallister OK, I will pull in and test. Probably won't merge it tonight. 
@aequasi if you have any issues with this pull request let me know.

jasonmccallister committed:
Add PSR to attempt array

ericvanjohnson commented:
New theme looks really good @themccallister. Not going to hold up this merge but if you could work on getting the footer to either go to the baseline or center I think that would look a lot better.

![group_topics_and_1__bash](https://cloud.githubusercontent.com/assets/89408/2807975/91df96c8-cd04-11e3-8499-e23894771e27.jpg)

jasonmccallister commented:
Thanks, I have ALOT of work left in that theme still.. I wanted to get some basics down so I wasn't looking at bare bootstrap styling. I'm going to make a new branch dedicated to the design when the core functionality is improved.
